### PR TITLE
ImportPlugin

### PR DESCRIPTION
**Pull request by thiloSchlemmer**
PlugIn to import vectorGeoResources, which are provided via Drag&Drop, FileInput etc.

### CODE REVIEW

File: src/plugins/ImportPlugin.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	 * @param {string} mimeType the mimeType of the data
+	 * @returns {GeoResource|null} the imported GeoResource or null on failure
+	 */
+	_importByData(data, mimeType) {
+		const sourceType = this._mapMimeTypeToVectorSourceType(mimeType);
+		const vectorGeoResource = this._importVectorDataService.importVectorData(data, { sourceType: sourceType });
+		if (vectorGeoResource) {
+			return vectorGeoResource;
+		}
+		emitNotification(this._translationService.translate('importPlugin_data_failed'), LevelTypes.ERROR);
```
Comment (taulinger) [inline src/plugins/ImportPlugin.js:89]: see above note on feedback

File: src/plugins/ImportPlugin.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	 */
+	async _importByUrl(url) {
+		const sourceType = await this._sourceTypeService.forURL(url);
+		const createGeoResource = (url, sourceType) => {
+			if (sourceType) {
+				switch (sourceType.name) {
+					case SourceTypeName.KML:
+					case SourceTypeName.GPX:
+					case SourceTypeName.GEOJSON:
+						return this._importVectorDataService.importVectorDataFromUrl(url);
```
Comment (taulinger) [inline src/plugins/ImportPlugin.js:61]: Here we have to set the `VectorSourceType` within the options => map sourceType to vectorSourceType

**Review by taulinger — COMMENTED** [review]
That's it 👌

File: src/plugins/ImportPlugin.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+				return sourceType.name === SourceTypeName.WMS ? warnOnRasterSourceType() : this._importVectorDataService.importVectorDataFromUrl(url, { sourceType: this._mapSourceTypeToVectorSourceType(sourceType) });
+			}
+			return null;
+		};
+
+		const geoResource = createGeoResource(url, sourceType);
+		if (geoResource) {
+			geoResource.onReject(() => {
+				console.warn(`Import failed for ${url}`);
+				emitNotification(`${this._translationService.translate('importPlugin_url_failed')}:${url}`, LevelTypes.WARN);
```
Comment (taulinger) [inline src/plugins/ImportPlugin.js:70]: user generated content -> XSS !

**Comment by thiloSchlemmer** [comment]
resolved by #730

**Pull request closed without merging**